### ISSUE TITLE
Numpy types written into bad pixel file

### ISSUE THREAD

**Issue by FlorianDoerr** [viperleed/viperleed#251]
An issue encountered by @S-Takezawa. It seems that with Numpy 2.1.1 the Numpy data types are written into the bad pixel file rather than the values.

**Comment by michele-riva** [comment]
Looks like https://github.com/numpy/numpy/pull/22449 is the cause

**Comment by michele-riva** [comment]
After discussing with @FlorianDoerr, we realized that the current fix in #252 is problematic for older numpy versions that may not have the expected `legacy` kwarg (or not accept `'1.25'` as a valid value). A good solution would be the following:

**viperleed/__init__.py**
```python
...
import numpy as np

NUMPY2_OR_LATER = np.__version__ >= '2'
...
```

**affected_module.py**
```python
from contextlib import nullcontext
...

from viperleed import NUMPY2_OR_LATER

...
print_scalar_as_number = (
    np.printoptions(legacy='1.25') if NUMPY2_OR_LATER
    else nullcontext()
    )

with print_scalar_as_number:
    ...
```

**Comment by FlorianDoerr** [comment]
The changes suggested by @michele-riva have been implemented in #254
